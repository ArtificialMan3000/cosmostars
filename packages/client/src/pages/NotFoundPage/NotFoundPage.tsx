import { Box, Button, Typography } from "@mui/material";
import { FC } from "react";
import { useNavigate } from "react-router-dom";

import { BasicLayout } from "@/shared/layouts/BasicLayout";
export const NotFoundPage: FC = () => {
  const navigate = useNavigate();
  const handleNavigate = () => {
    navigate(-1);
  };
  return (
    <BasicLayout>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          alignItems: "center",
          my: "auto",
          height: "70%",
          width: "100%",
        }}>
        <Typography
          component="h1"
          className="main-page__name"
          sx={{
            fontSize: "144px",
            letterSpacing: "0.01em",
            color: "transparent",
            textShadow:
              " 5px 4px 6px #011133, 0px 0px 0px #ff00008c, 5px 4px 6px #ff00008c",
            marginBottom: 1,
            WebkitTextStroke: "4px white",
            pointerEvents: "none",
          }}>
          Oops...Error!
        </Typography>
        <Button variant="contained" size="large" onClick={handleNavigate}>
          Back
        </Button>
      </Box>
    </BasicLayout>
  );
};
