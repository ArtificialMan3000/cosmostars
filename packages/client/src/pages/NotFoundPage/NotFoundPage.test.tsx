import { render, screen } from "@testing-library/react";
import { NotFoundPage } from "./NotFoundPage";

describe("NotFound Page", () => {
  it("should render without error", () => {
    expect(() => render(<NotFoundPage />)).not.toThrowError();
  });

  it("should have correct header", () => {
    const header = "404";

    render(<NotFoundPage />);
    const headerElem = screen.getByText(header, { exact: false });

    expect(headerElem).toBeDefined();
    expect(headerElem).toHaveProperty("tagName", "H1");
  });
});
