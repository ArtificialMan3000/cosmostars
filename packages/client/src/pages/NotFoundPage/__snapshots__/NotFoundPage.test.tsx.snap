// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-wlq4rs"
  >
    <div
      class="MuiBox-root css-5jrfta"
    >
      <h1
        class="MuiTypography-root MuiTypography-body1 main-page__name css-v6tnx3-MuiTypography-root"
      >
        Oops...Error!
      </h1>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
